src/runtime/FunctionObject.cpp

```diff
@@ -320,6 +320,131 @@ Object* FunctionObject::newInstance(ExecutionState& state, const size_t argc, Va
         return receiver;
 }
 
+Value FunctionObject::processFastCall(ExecutionState& state, const Value& receiverSrc, const size_t argc, Value* argv, bool isNewExpression)
+{
+    Context* ctx = m_codeBlock->context();
+    bool isStrict = m_codeBlock->isStrict();
+
+#ifndef NDEBUG
+    bool isSuperCall = state.executionContext() != nullptr ? state.executionContext()->isOnGoingSuperCall() : false;
+    ASSERT(isSuperCall == false);
+#endif
+
+    if (m_codeBlock->isInterpretedCodeBlock() == false) {
+        CallNativeFunctionData* code = m_codeBlock->nativeFunctionData();
+        FunctionEnvironmentRecordSimple record(this);
+        LexicalEnvironment env(&record, outerEnvironment());
+        ExecutionContext ec(ctx, state.executionContext(), &env, isStrict);
+
+        size_t len = m_codeBlock->parameterCount();
+        if (argc < len) {
+            Value* newArgv = (Value*)alloca(sizeof(Value) * len);
+            for (size_t i = 0; i < argc; i++) {
+                newArgv[i] = argv[i];
+            }
+            for (size_t i = argc; i < len; i++) {
+                newArgv[i] = Value();
+            }
+            argv = newArgv;
+        }
+
+        Value receiver = receiverSrc;
+        // prepare receiver
+        if (UNLIKELY(isStrict == false)) {
+            if (receiver.isUndefinedOrNull()) {
+                receiver = ctx->globalObject();
+            } else {
+                receiver = receiver.toObject(state);
+            }
+        }
+
+        record.bindThisValue(state, receiver);
+        if (receiverSrc.isObject()) {
```

Comment:
For readability / State expression explicitly
if (receiverSrc.isObject() == true)